internal/exec/stages/files/passwd.go

```diff
@@ -128,9 +130,11 @@ func (s stage) createGroups(config types.Config) error {
 	defer s.Logger.PopPrefix()
 
 	for _, g := range config.Passwd.Groups {
-		if err := s.CreateGroup(g); err != nil {
+		if err := s.EnsureGroup(g); err != nil {
 			return fmt.Errorf("failed to create group %q: %v",
 				g.Name, err)
+		} else if !(g.ShouldExist == nil || *g.ShouldExist) && err == nil {
+			continue
```

Comment:
This doesn't seem necessary, there aren't extra actions after ensuring the group exists like there are with users.